datastream-cloud-storage/src/main/java/com/linkedin/datastream/cloud/storage/io/ScribeParquetFile.java

```diff
@@ -0,0 +1,92 @@
+/**
+ *  Copyright 2020 Wayfair LLC. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.cloud.storage.io;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaBuilder;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.avro.AvroParquetWriter;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linkedin.datastream.cloud.storage.Package;
+import com.linkedin.datastream.common.VerifiableProperties;
+
+/**
+ * Implementation of {@link com.linkedin.datastream.cloud.storage.io.File} to support parquet file format for scribe
+ */
+public class ScribeParquetFile implements com.linkedin.datastream.cloud.storage.io.File {
+    private static final Logger LOG = LoggerFactory.getLogger(AvroParquetFile.class);
+    private static final String CONFIG_PAGE_SIZE = "pageSize";
+    private static final int DEFAULT_PAGE_SIZE = 64 * 1024;
```

Comment:
You might want to sync Parquet `pageSize` with the `maxFileSize` in `ObjectBuilder`. If I remember correctly, having too many pages in a Parquet file results in performance degradation.

Reply:
On second thought, it might be Parquet RowGroups number that directly affect the performance. I'd still double-check this though just to make sure we are not accidentally causing performance issues.